src/lex.cpp

```diff
@@ -759,6 +759,7 @@ char8* lexer::check_garbage_in_number_literal(char8* input) {
     switch (*input) {
     QLJS_CASE_DECIMAL_DIGIT:
     QLJS_CASE_IDENTIFIER_START:
+    case u8'.':
```

Comment:
Would this change also affect hex, octal, and decimal literals? Should we test those as well?

Reply:
This function was affecting both binary and hexadecimal cases not the octal case. 
There's exact code path in the octal case so I made it polymorphic over `QLJS_ERROR_TYPE` 
in commit  83f70d13eaa579a6a13cbe24478d24cf96b4d64e using template parameter.

Reply:
And We have simple tests for all the hex, octal and decimal literals. octal test case has been there already.

# new test cases links
- `"0b1.1"` [binary test case](https://github.com/catac1/quick-lint-js/blob/1538d2a80a24032d71d70e17a35e9d2852acbf25/test/test-lex.cpp#L174)
- `"0xf.f"` [hex test case](https://github.com/catac1/quick-lint-js/blob/1538d2a80a24032d71d70e17a35e9d2852acbf25/test/test-lex.cpp#L265)

this is very simplistic test. so we have to add more.